aws-cpp-sdk-core/source/client/AWSClient.cpp

```diff
@@ -707,7 +707,10 @@ void AWSClient::AddChecksumToRequest(const std::shared_ptr<Aws::Http::HttpReques
         }
         else if (checksumAlgorithmName == "md5")
         {
-            httpRequest->SetHeaderValue(Http::CONTENT_MD5_HEADER, HashingUtils::Base64Encode(HashingUtils::CalculateMD5(*(request.GetBody()))));
+            if (request.GetBody())
```

Comment:
hi,
first of all, could you please also fix this nullptr deference issue in this file for other hash algorithms?
secondly, did you check how it is going to work by actually calling s3?
I would try setting md5("") (md5 of nothing) value if user tries to set an empty body.